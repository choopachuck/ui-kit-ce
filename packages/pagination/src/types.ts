import { UsePaginationProps, PaginationItem } from './hooks'
import { PaginationClasses } from './classes'
import { ElementSizeType, ComponentPropsWithRefFix } from '@v-uik/common'
import { PaginationComponentsConfig } from './components'

export type PaginationProps = ComponentPropsWithRefFix<'nav'> &
  UsePaginationProps & {
    /**
     * CSS классы компонента
     */
    classes?: PaginationClasses
    /**
     * Размер элементов пагинации
     */
    size?: ElementSizeType
    /**
     * Состояние блокировки элементов пагинации
     */
    disabled?: boolean
    /**
     * Свойство для переопределения элементов `Pagination`
     */
    components?: PaginationComponentsConfig
    /**
     * Флаг скрытия кнопки навигации к первой странице
     */
    showNavigationFirst?: boolean
    /**
     * Флаг скрытия кнопки навигации к последней странице
     */
    showNavigationLast?: boolean
    /**
     * Флаг скрытия кнопки навигации к предыдущей странице
     */
    hideNavigationPrevious?: boolean
    /**
     * Флаг скрытия кнопки навигации к следующей странице
     */
    hideNavigationNext?: boolean
    /**
     * Свойства для кнопки навигации к первой странице
     */
    firstButtonProps?: ComponentPropsWithRefFix<'button'>
    /**
     * Свойства для кнопки навигации к первой странице
     */
    lastButtonProps?: ComponentPropsWithRefFix<'button'>
    /**
     * Свойства для кнопки навигации к предыдущей странице
     */
    previousButtonProps?: ComponentPropsWithRefFix<'button'>
    /**
     * Свойства для кнопки навигации к следующей странице
     */
    nextButtonProps?: ComponentPropsWithRefFix<'button'>
    /**
     * Функция для получения свойства для каждого скрытого элемента
     */
    getPaginationOverflowButtonProps?: (
      item: PaginationItem
    ) => ComponentPropsWithRefFix<'button'>
    /**
     * Функция для получения свойства для каждого элемента страницы
     */
    getPaginationPageButtonProps?: (
      item: PaginationItem
    ) => ComponentPropsWithRefFix<'button'>
  }
