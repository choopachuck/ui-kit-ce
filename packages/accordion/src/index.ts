export * from './Accordion'
export * from './AccordionItem'
export type {
  AccordionIconProps,
  AccordionComponentsConfig,
} from './components'
