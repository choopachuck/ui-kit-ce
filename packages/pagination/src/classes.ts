export type PaginationClasses = {
  /** Стиль, применяемый к основному элементу пагинации */
  root?: string
  /** Стиль, применяемый ко всем элементам пагинации */
  item?: string
  /** Стиль, применяемый ко всем элементам пагинации с `size="sm"` */
  itemSmall?: string
  /** Стиль, применяемый ко всем элементам пагинации с `size="md"` */
  itemMedium?: string
  /** Стиль, применяемый ко всем элементам пагинации с `size="lg"` */
  itemLarge?: string
  /** Стиль, применяемый ко всем неактивным элементам пагинации */
  itemDisabled?: string
  /** Стиль, применяемый к активному элементу пагинации */
  itemSelected?: string
  /** Стиль, применяемый к индикатору активного элемента */
  selectedIndicator?: string
  /** Стиль, применяемый к индикатору активного элемента с `size="sm"` */
  selectedIndicatorSmall?: string
  /** Стиль, применяемый к индикатору активного элемента с `size="md"` */
  selectedIndicatorMedium?: string
  /** Стиль, применяемый к индикатору активного элемента с `size="lg"` */
  selectedIndicatorLarge?: string
  /** Стиль, применяемый к индикатору заблокированного активного элемента */
  selectedIndicatorDisabled?: string
  /** Стиль, применяемый ко всем скрытым элементам пагинации */
  itemOverflow?: string
  /** Стиль, применяемый ко всем навигационным кнопкам пагинации */
  navigationButton?: string
  /** Стиль, применяемый к навигационной кнопке предыдущей страницы */
  navigationButtonPrevious?: string
  /** Стиль, применяемый к навигационной кнопке следующей страницы  */
  navigationButtonNext?: string
  /** Стиль, применяемый к навигационной кнопке первой страницы */
  navigationButtonFirst?: string
  /** Стиль, применяемый к навигационной кнопке последней страницы  */
  navigationButtonLast?: string
}
