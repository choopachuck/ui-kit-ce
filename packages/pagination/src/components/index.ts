import {
  PaginationPageButton,
  PaginationPageButtonProps,
} from './PaginationPageButton'
import {
  PaginationOverflowButton,
  PaginationOverflowButtonProps,
} from './PaginationOverflowButton'
import {
  PaginationNavigationButton,
  PaginationNavigationButtonProps,
} from './PaginationNavigationButton'
import {
  PaginationSelectedIndicator,
  PaginationSelectedIndicatorProps,
} from './PaginationSelectedIndicator'

export type PaginationComponents = {
  PageButton: React.ComponentType<PaginationPageButtonProps>
  OverflowButton: React.ComponentType<PaginationOverflowButtonProps>
  PreviousPageButton: React.ComponentType<PaginationNavigationButtonProps>
  NextPageButton: React.ComponentType<PaginationNavigationButtonProps>
  SelectedIndicator: React.ComponentType<PaginationSelectedIndicatorProps>
  FirstPageButton: React.ComponentType<PaginationNavigationButtonProps>
  LastPageButton: React.ComponentType<PaginationNavigationButtonProps>
}

const unionComponents = (
  generic: PaginationComponents,
  custom?: PaginationComponentsConfig
) => {
  return (Object.keys(generic) as (keyof PaginationComponents)[]).reduce(
    (accum, componentName) => {
      if (custom?.[componentName]) {
        accum[componentName] = custom[
          componentName
        ] as PaginationComponents[keyof PaginationComponents]
      } else {
        accum[componentName] = generic[componentName]
      }

      return accum
    },
    {} as PaginationComponents
  )
}
export type PaginationComponentsConfig = Partial<PaginationComponents>

export const defaultPaginationComponents = {
  PageButton: PaginationPageButton,
  OverflowButton: PaginationOverflowButton,
  PreviousPageButton: PaginationNavigationButton,
  NextPageButton: PaginationNavigationButton,
  SelectedIndicator: PaginationSelectedIndicator,
  FirstPageButton: PaginationNavigationButton,
  LastPageButton: PaginationNavigationButton,
}

export type PaginationComponentsGeneric = typeof defaultPaginationComponents

export const getComponents = (
  componentProp?: PaginationComponentsConfig | undefined
): PaginationComponentsGeneric =>
  unionComponents(
    defaultPaginationComponents,
    componentProp
  ) as PaginationComponentsGeneric

export type {
  PaginationPageButtonProps,
  PaginationOverflowButtonProps,
  PaginationNavigationButtonProps,
  PaginationSelectedIndicatorProps,
}
