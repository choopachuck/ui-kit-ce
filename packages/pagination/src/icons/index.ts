export * from './PaginationArrowIcon'
export * from './PaginationArrowLastIcon'
