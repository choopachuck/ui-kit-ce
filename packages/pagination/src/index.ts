export * from './Pagination'
export * from './hooks'
export type {
  PaginationPageButtonProps,
  PaginationOverflowButtonProps,
  PaginationNavigationButtonProps,
  PaginationSelectedIndicatorProps,
  PaginationComponents,
  PaginationComponentsConfig,
  PaginationComponentsGeneric,
} from './components'
export * from './types'
export * from './classes'
