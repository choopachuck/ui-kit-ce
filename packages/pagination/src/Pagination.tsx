'use client'

/* eslint-disable react/prop-types */ // TODO: косячит eslint при типизации пропсов компонента через дженерики React.forwardRef

import * as React from 'react'
import { clsx } from '@v-uik/theme'
import { useClassList, useButtonReset } from '@v-uik/hooks'
import { isPaginationItemOverflow, usePagination } from './hooks'
import { ElementSize } from '@v-uik/common'
import { PaginationArrowIcon, PaginationArrowLastIcon } from './icons'
import { getComponents } from './components'
import { PaginationProps } from './types'
import { useStyles } from './styles'

export const Pagination = React.forwardRef<HTMLDivElement, PaginationProps>(
  (
    {
      classes: classesProp,
      totalPageCount = 1,
      nearPageCount = 1,
      boundaryPageCount = 1,
      currentPage: currentPageProp,
      defaultPage,
      size = 'md',
      disabled = false,
      components,
      className,
      showNavigationFirst = false,
      showNavigationLast = false,
      hideNavigationPrevious = false,
      hideNavigationNext = false,
      onPageChange,
      nextButtonProps,
      previousButtonProps,
      lastButtonProps,
      firstButtonProps,
      getPaginationOverflowButtonProps,
      getPaginationPageButtonProps,
      ...rest
    },
    ref
  ) => {
    const classes = useStyles()
    const classesList = useClassList(classes, classesProp)
    const buttonClasses = useButtonReset()
    const classesMap = {
      ...classesList,
      item: clsx(buttonClasses.resetButton, classesList.item),
    }

    const {
      PageButton,
      OverflowButton,
      SelectedIndicator,
      FirstPageButton,
      LastPageButton,
      NextPageButton,
      PreviousPageButton,
    } = getComponents(components)

    const {
      items,
      isEnd,
      isStart,
      setCurrentPage,
      handleFirstClick,
      handleLastClick,
      handleNextClick,
      handlePreviousClick,
    } = usePagination({
      boundaryPageCount,
      currentPage: currentPageProp,
      nearPageCount,
      totalPageCount,
      defaultPage,
      onPageChange,
    })

    const makeHandleButtonClick = (pageNumber: number) => () => {
      setCurrentPage(pageNumber)
    }

    const isSmall = size === ElementSize.sm
    const isMedium = size === ElementSize.md
    const isLarge = size === ElementSize.lg

    const isNavigationPrevOrFirstDisabled =
      disabled || isStart || !totalPageCount
    const isNavigationNextOrLastDisabled = disabled || isEnd || !totalPageCount

    const paginationSelectedIndicatorClassName = clsx(
      classesMap.selectedIndicator,
      {
        [classesMap.selectedIndicatorDisabled]: disabled,
        [classesMap.selectedIndicatorSmall]: isSmall,
        [classesMap.selectedIndicatorMedium]: isMedium,
        [classesMap.selectedIndicatorLarge]: isLarge,
      }
    )

    const itemClassName = clsx(classesMap.item, {
      [classesMap.itemDisabled]: disabled,
      [classesMap.itemSmall]: size === ElementSize.sm,
      [classesMap.itemMedium]: size === ElementSize.md,
      [classesMap.itemLarge]: size === ElementSize.lg,
    })

    return (
      <nav
        ref={ref}
        aria-label="pagination"
        {...rest}
        className={clsx(classesMap.root, className)}
      >
        {showNavigationFirst && (
          <FirstPageButton
            aria-label="Go to first page"
            {...firstButtonProps}
            disabled={
              firstButtonProps?.disabled || isNavigationPrevOrFirstDisabled
            }
            isStart={isStart}
            isEnd={isEnd}
            size={size}
            className={clsx(
              itemClassName,
              classesMap.navigationButton,
              classesMap.navigationButtonFirst,
              firstButtonProps?.className,
              {
                [classesMap.itemDisabled]: isNavigationPrevOrFirstDisabled,
              }
            )}
            onPageChange={handleFirstClick}
          >
            <PaginationArrowLastIcon />
          </FirstPageButton>
        )}
        {!hideNavigationPrevious && (
          <PreviousPageButton
            aria-label="Go to previous page"
            {...previousButtonProps}
            disabled={
              previousButtonProps?.disabled || isNavigationPrevOrFirstDisabled
            }
            isStart={isStart}
            isEnd={isEnd}
            size={size}
            className={clsx(
              itemClassName,
              classesMap.navigationButton,
              classesMap.navigationButtonPrevious,
              previousButtonProps?.className,
              {
                [classesMap.itemDisabled]: isNavigationPrevOrFirstDisabled,
              }
            )}
            onPageChange={handlePreviousClick}
          >
            <PaginationArrowIcon />
          </PreviousPageButton>
        )}
        {Array.from({ length: items?.length }).map((_, i) => {
          const pageNumber = items[i].value
          const isSelected = items[i].isSelected
          const isOverflow = isPaginationItemOverflow(pageNumber)

          const nativeProps =
            (isOverflow
              ? getPaginationOverflowButtonProps?.(items[i])
              : getPaginationPageButtonProps?.(items[i])) || {}

          const baseProps = {
            children: pageNumber,
            disabled: nativeProps?.disabled || disabled,
            size,
            appearance: items[i].appearance,
          }

          if (isPaginationItemOverflow(pageNumber)) {
            return (
              <OverflowButton
                {...nativeProps}
                {...baseProps}
                key={`pagination-overflow-${i}`}
                className={clsx(
                  itemClassName,
                  classesMap.itemOverflow,
                  nativeProps?.className
                )}
              />
            )
          }

          return (
            <PageButton
              aria-label={`Go to page ${pageNumber}`}
              {...nativeProps}
              {...baseProps}
              key={`pagination-page-${i}`}
              isSelected={isSelected}
              className={clsx(
                itemClassName,
                classesMap.item,
                nativeProps?.className,
                {
                  [classesMap.itemSelected]: isSelected,
                }
              )}
              indicator={
                <SelectedIndicator
                  className={paginationSelectedIndicatorClassName}
                />
              }
              onPageChange={makeHandleButtonClick(Number(pageNumber))}
            />
          )
        })}
        {!hideNavigationNext && (
          <NextPageButton
            aria-label="Go to next page"
            {...nextButtonProps}
            isStart={isStart}
            isEnd={isEnd}
            size={size}
            disabled={
              nextButtonProps?.disabled || isNavigationNextOrLastDisabled
            }
            className={clsx(
              itemClassName,
              classesMap.navigationButton,
              classesMap.navigationButtonNext,
              nextButtonProps?.className,
              {
                [classesMap.itemDisabled]: isNavigationNextOrLastDisabled,
              }
            )}
            onPageChange={handleNextClick}
          >
            <PaginationArrowIcon />
          </NextPageButton>
        )}
        {showNavigationLast && (
          <LastPageButton
            aria-label="Go to last page"
            {...lastButtonProps}
            isStart={isStart}
            isEnd={isEnd}
            size={size}
            disabled={
              lastButtonProps?.disabled || isNavigationNextOrLastDisabled
            }
            className={clsx(
              itemClassName,
              classesMap.navigationButton,
              classesMap.navigationButtonLast,
              lastButtonProps?.className,
              {
                [classesMap.itemDisabled]: isNavigationNextOrLastDisabled,
              }
            )}
            onPageChange={handleLastClick}
          >
            <PaginationArrowLastIcon />
          </LastPageButton>
        )}
      </nav>
    )
  }
)
