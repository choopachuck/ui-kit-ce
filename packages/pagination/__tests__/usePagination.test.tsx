import * as React from 'react'
import { render, fireEvent } from '@testing-library/react'
import { UsePaginationReturnProps } from '../src'
import { UsePaginationComponent } from './testComponents/UsePaginationComponent'
import {
  usePaginationTestProps02,
  usePaginationTestProps01,
  BUTTONS,
} from './testComponents/usePaginationTestData'

it.each(usePaginationTestProps01)(
  'use pagination returns props correctly (%p)',
  (props) => {
    let returnProps = {}
    const handleGetProps = (props: UsePaginationReturnProps) => {
      returnProps = props
    }

    const { getByText } = render(
      <UsePaginationComponent getProps={handleGetProps} {...props} />
    )

    fireEvent.click(getByText(BUTTONS.GET_PROPS))

    expect(returnProps).toMatchSnapshot()
  }
)

it.each(usePaginationTestProps02)(
  'use pagination handleChangeCurrentPage works correctly (%p)',
  (props) => {
    let returnProps = {}
    const handleGetProps = (props: UsePaginationReturnProps) => {
      returnProps = props
    }
    const { getByText } = render(
      <UsePaginationComponent
        getProps={handleGetProps}
        totalPageCount={25}
        handleChangeCurrentPageValue={props.pageNumber}
        defaultPage={props.defaultPage}
      />
    )

    fireEvent.click(getByText(BUTTONS.HANDLE_CHANGE_CURRENT_PAGE))

    fireEvent.click(getByText(BUTTONS.GET_PROPS))

    expect(returnProps).toMatchSnapshot()
  }
)

it('use pagination handleFirstClick works correctly (%p)', () => {
  let returnProps = {}
  const handleGetProps = (props: UsePaginationReturnProps) => {
    returnProps = props
  }
  const { getByText } = render(
    <UsePaginationComponent
      getProps={handleGetProps}
      totalPageCount={25}
      defaultPage={5}
    />
  )

  fireEvent.click(getByText(BUTTONS.HANDLE_FIRST_CLICK))

  fireEvent.click(getByText(BUTTONS.GET_PROPS))

  expect(returnProps).toMatchSnapshot()
})

it('use pagination handleLastClick works correctly (%p)', () => {
  let returnProps = {}
  const handleGetProps = (props: UsePaginationReturnProps) => {
    returnProps = props
  }
  const { getByText } = render(
    <UsePaginationComponent
      getProps={handleGetProps}
      totalPageCount={25}
      defaultPage={5}
    />
  )

  fireEvent.click(getByText(BUTTONS.HANDLE_LAST_CLICK))

  fireEvent.click(getByText(BUTTONS.GET_PROPS))

  expect(returnProps).toMatchSnapshot()
})

it('use pagination handleNextClick works correctly (%p)', () => {
  let returnProps = {}
  const handleGetProps = (props: UsePaginationReturnProps) => {
    returnProps = props
  }
  const { getByText } = render(
    <UsePaginationComponent
      getProps={handleGetProps}
      totalPageCount={25}
      defaultPage={5}
    />
  )

  fireEvent.click(getByText(BUTTONS.HANDLE_NEXT_CLICK))

  fireEvent.click(getByText(BUTTONS.GET_PROPS))

  expect(returnProps).toMatchSnapshot()
})

it('use pagination handlePreviousClick works correctly (%p)', () => {
  let returnProps = {}
  const handleGetProps = (props: UsePaginationReturnProps) => {
    returnProps = props
  }
  const { getByText } = render(
    <UsePaginationComponent
      getProps={handleGetProps}
      totalPageCount={25}
      defaultPage={5}
    />
  )

  fireEvent.click(getByText(BUTTONS.HANDLE_PREVIOUS_CLICK))

  fireEvent.click(getByText(BUTTONS.GET_PROPS))

  expect(returnProps).toMatchSnapshot()
})
