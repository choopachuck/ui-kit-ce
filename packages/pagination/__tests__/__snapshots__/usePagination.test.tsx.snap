// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`use pagination handleChangeCurrentPage works correctly ({"defaultPage": 8, "pageNumber": 2}) 1`] = `
Object {
  "currentPage": 2,
  "handleFirstClick": [Function],
  "handleLastClick": [Function],
  "handleNextClick": [Function],
  "handlePreviousClick": [Function],
  "isEnd": false,
  "isStart": false,
  "items": Array [
    Object {
      "appearance": "item",
      "isSelected": false,
      "value": 1,
    },
    Object {
      "appearance": "item",
      "isSelected": true,
      "value": 2,
    },
    Object {
      "appearance": "item",
      "isSelected": false,
      "value": 3,
    },
    Object {
      "appearance": "item",
      "isSelected": false,
      "value": 4,
    },
    Object {
      "appearance": "item",
      "isSelected": false,
      "value": 5,
    },
    Object {
      "appearance": "overflow-end",
      "isSelected": false,
      "value": "...",
    },
    Object {
      "appearance": "item",
      "isSelected": false,
      "value": 25,
    },
  ],
  "setCurrentPage": [Function],
}
`;

exports[`use pagination handleChangeCurrentPage works correctly ({"pageNumber": 2}) 1`] = `
Object {
  "currentPage": 2,
  "handleFirstClick": [Function],
  "handleLastClick": [Function],
  "handleNextClick": [Function],
  "handlePreviousClick": [Function],
  "isEnd": false,
  "isStart": false,
  "items": Array [
    Object {
      "appearance": "item",
      "isSelected": false,
      "value": 1,
    },
    Object {
      "appearance": "item",
      "isSelected": true,
      "value": 2,
    },
    Object {
      "appearance": "item",
      "isSelected": false,
      "value": 3,
    },
    Object {
      "appearance": "item",
      "isSelected": false,
      "value": 4,
    },
    Object {
      "appearance": "item",
      "isSelected": false,
      "value": 5,
    },
    Object {
      "appearance": "overflow-end",
      "isSelected": false,
      "value": "...",
    },
    Object {
      "appearance": "item",
      "isSelected": false,
      "value": 25,
    },
  ],
  "setCurrentPage": [Function],
}
`;

exports[`use pagination handleChangeCurrentPage works correctly ({"pageNumber": 25}) 1`] = `
Object {
  "currentPage": 25,
  "handleFirstClick": [Function],
  "handleLastClick": [Function],
  "handleNextClick": [Function],
  "handlePreviousClick": [Function],
  "isEnd": true,
  "isStart": false,
  "items": Array [
    Object {
      "appearance": "item",
      "isSelected": false,
      "value": 1,
    },
    Object {
      "appearance": "overflow-start",
      "isSelected": false,
      "value": "...",
    },
    Object {
      "appearance": "item",
      "isSelected": false,
      "value": 21,
    },
    Object {
      "appearance": "item",
      "isSelected": false,
      "value": 22,
    },
    Object {
      "appearance": "item",
      "isSelected": false,
      "value": 23,
    },
    Object {
      "appearance": "item",
      "isSelected": false,
      "value": 24,
    },
    Object {
      "appearance": "item",
      "isSelected": true,
      "value": 25,
    },
  ],
  "setCurrentPage": [Function],
}
`;

exports[`use pagination handleFirstClick works correctly (%p) 1`] = `
Object {
  "currentPage": 1,
  "handleFirstClick": [Function],
  "handleLastClick": [Function],
  "handleNextClick": [Function],
  "handlePreviousClick": [Function],
  "isEnd": false,
  "isStart": true,
  "items": Array [
    Object {
      "appearance": "item",
      "isSelected": true,
      "value": 1,
    },
    Object {
      "appearance": "item",
      "isSelected": false,
      "value": 2,
    },
    Object {
      "appearance": "item",
      "isSelected": false,
      "value": 3,
    },
    Object {
      "appearance": "item",
      "isSelected": false,
      "value": 4,
    },
    Object {
      "appearance": "item",
      "isSelected": false,
      "value": 5,
    },
    Object {
      "appearance": "overflow-end",
      "isSelected": false,
      "value": "...",
    },
    Object {
      "appearance": "item",
      "isSelected": false,
      "value": 25,
    },
  ],
  "setCurrentPage": [Function],
}
`;

exports[`use pagination handleLastClick works correctly (%p) 1`] = `
Object {
  "currentPage": 25,
  "handleFirstClick": [Function],
  "handleLastClick": [Function],
  "handleNextClick": [Function],
  "handlePreviousClick": [Function],
  "isEnd": true,
  "isStart": false,
  "items": Array [
    Object {
      "appearance": "item",
      "isSelected": false,
      "value": 1,
    },
    Object {
      "appearance": "overflow-start",
      "isSelected": false,
      "value": "...",
    },
    Object {
      "appearance": "item",
      "isSelected": false,
      "value": 21,
    },
    Object {
      "appearance": "item",
      "isSelected": false,
      "value": 22,
    },
    Object {
      "appearance": "item",
      "isSelected": false,
      "value": 23,
    },
    Object {
      "appearance": "item",
      "isSelected": false,
      "value": 24,
    },
    Object {
      "appearance": "item",
      "isSelected": true,
      "value": 25,
    },
  ],
  "setCurrentPage": [Function],
}
`;

exports[`use pagination handleNextClick works correctly (%p) 1`] = `
Object {
  "currentPage": 6,
  "handleFirstClick": [Function],
  "handleLastClick": [Function],
  "handleNextClick": [Function],
  "handlePreviousClick": [Function],
  "isEnd": false,
  "isStart": false,
  "items": Array [
    Object {
      "appearance": "item",
      "isSelected": false,
      "value": 1,
    },
    Object {
      "appearance": "overflow-start",
      "isSelected": false,
      "value": "...",
    },
    Object {
      "appearance": "item",
      "isSelected": false,
      "value": 5,
    },
    Object {
      "appearance": "item",
      "isSelected": true,
      "value": 6,
    },
    Object {
      "appearance": "item",
      "isSelected": false,
      "value": 7,
    },
    Object {
      "appearance": "overflow-end",
      "isSelected": false,
      "value": "...",
    },
    Object {
      "appearance": "item",
      "isSelected": false,
      "value": 25,
    },
  ],
  "setCurrentPage": [Function],
}
`;

exports[`use pagination handlePreviousClick works correctly (%p) 1`] = `
Object {
  "currentPage": 4,
  "handleFirstClick": [Function],
  "handleLastClick": [Function],
  "handleNextClick": [Function],
  "handlePreviousClick": [Function],
  "isEnd": false,
  "isStart": false,
  "items": Array [
    Object {
      "appearance": "item",
      "isSelected": false,
      "value": 1,
    },
    Object {
      "appearance": "overflow-start",
      "isSelected": false,
      "value": "...",
    },
    Object {
      "appearance": "item",
      "isSelected": false,
      "value": 3,
    },
    Object {
      "appearance": "item",
      "isSelected": true,
      "value": 4,
    },
    Object {
      "appearance": "item",
      "isSelected": false,
      "value": 5,
    },
    Object {
      "appearance": "overflow-end",
      "isSelected": false,
      "value": "...",
    },
    Object {
      "appearance": "item",
      "isSelected": false,
      "value": 25,
    },
  ],
  "setCurrentPage": [Function],
}
`;

exports[`use pagination returns props correctly ({"boundaryPageCount": 0, "defaultPage": 3, "nearPageCount": 1, "totalPageCount": 10}) 1`] = `
Object {
  "currentPage": 3,
  "handleFirstClick": [Function],
  "handleLastClick": [Function],
  "handleNextClick": [Function],
  "handlePreviousClick": [Function],
  "isEnd": false,
  "isStart": false,
  "items": Array [
    Object {
      "appearance": "overflow-start",
      "isSelected": false,
      "value": "...",
    },
    Object {
      "appearance": "item",
      "isSelected": false,
      "value": 2,
    },
    Object {
      "appearance": "item",
      "isSelected": true,
      "value": 3,
    },
    Object {
      "appearance": "item",
      "isSelected": false,
      "value": 4,
    },
    Object {
      "appearance": "overflow-end",
      "isSelected": false,
      "value": "...",
    },
  ],
  "setCurrentPage": [Function],
}
`;

exports[`use pagination returns props correctly ({"boundaryPageCount": 0, "nearPageCount": 1, "totalPageCount": 10}) 1`] = `
Object {
  "currentPage": 1,
  "handleFirstClick": [Function],
  "handleLastClick": [Function],
  "handleNextClick": [Function],
  "handlePreviousClick": [Function],
  "isEnd": false,
  "isStart": true,
  "items": Array [
    Object {
      "appearance": "item",
      "isSelected": true,
      "value": 1,
    },
    Object {
      "appearance": "item",
      "isSelected": false,
      "value": 2,
    },
    Object {
      "appearance": "item",
      "isSelected": false,
      "value": 3,
    },
    Object {
      "appearance": "item",
      "isSelected": false,
      "value": 4,
    },
    Object {
      "appearance": "overflow-end",
      "isSelected": false,
      "value": "...",
    },
  ],
  "setCurrentPage": [Function],
}
`;

exports[`use pagination returns props correctly ({"boundaryPageCount": 1, "currentPage": 1, "defaultPage": 3, "nearPageCount": 2, "totalPageCount": 8}) 1`] = `
Object {
  "currentPage": 1,
  "handleFirstClick": [Function],
  "handleLastClick": [Function],
  "handleNextClick": [Function],
  "handlePreviousClick": [Function],
  "isEnd": false,
  "isStart": true,
  "items": Array [
    Object {
      "appearance": "item",
      "isSelected": true,
      "value": 1,
    },
    Object {
      "appearance": "item",
      "isSelected": false,
      "value": 2,
    },
    Object {
      "appearance": "item",
      "isSelected": false,
      "value": 3,
    },
    Object {
      "appearance": "item",
      "isSelected": false,
      "value": 4,
    },
    Object {
      "appearance": "item",
      "isSelected": false,
      "value": 5,
    },
    Object {
      "appearance": "item",
      "isSelected": false,
      "value": 6,
    },
    Object {
      "appearance": "item",
      "isSelected": false,
      "value": 7,
    },
    Object {
      "appearance": "overflow-end",
      "isSelected": false,
      "value": "...",
    },
    Object {
      "appearance": "item",
      "isSelected": false,
      "value": 8,
    },
  ],
  "setCurrentPage": [Function],
}
`;

exports[`use pagination returns props correctly ({"boundaryPageCount": 3, "currentPage": 6, "nearPageCount": 2, "totalPageCount": 25}) 1`] = `
Object {
  "currentPage": 6,
  "handleFirstClick": [Function],
  "handleLastClick": [Function],
  "handleNextClick": [Function],
  "handlePreviousClick": [Function],
  "isEnd": false,
  "isStart": false,
  "items": Array [
    Object {
      "appearance": "item",
      "isSelected": false,
      "value": 1,
    },
    Object {
      "appearance": "item",
      "isSelected": false,
      "value": 2,
    },
    Object {
      "appearance": "item",
      "isSelected": false,
      "value": 3,
    },
    Object {
      "appearance": "item",
      "isSelected": false,
      "value": 4,
    },
    Object {
      "appearance": "item",
      "isSelected": false,
      "value": 5,
    },
    Object {
      "appearance": "item",
      "isSelected": true,
      "value": 6,
    },
    Object {
      "appearance": "item",
      "isSelected": false,
      "value": 7,
    },
    Object {
      "appearance": "item",
      "isSelected": false,
      "value": 8,
    },
    Object {
      "appearance": "item",
      "isSelected": false,
      "value": 9,
    },
    Object {
      "appearance": "overflow-end",
      "isSelected": false,
      "value": "...",
    },
    Object {
      "appearance": "item",
      "isSelected": false,
      "value": 23,
    },
    Object {
      "appearance": "item",
      "isSelected": false,
      "value": 24,
    },
    Object {
      "appearance": "item",
      "isSelected": false,
      "value": 25,
    },
  ],
  "setCurrentPage": [Function],
}
`;

exports[`use pagination returns props correctly ({"boundaryPageCount": 3, "nearPageCount": 2, "totalPageCount": 25}) 1`] = `
Object {
  "currentPage": 1,
  "handleFirstClick": [Function],
  "handleLastClick": [Function],
  "handleNextClick": [Function],
  "handlePreviousClick": [Function],
  "isEnd": false,
  "isStart": true,
  "items": Array [
    Object {
      "appearance": "item",
      "isSelected": true,
      "value": 1,
    },
    Object {
      "appearance": "item",
      "isSelected": false,
      "value": 2,
    },
    Object {
      "appearance": "item",
      "isSelected": false,
      "value": 3,
    },
    Object {
      "appearance": "item",
      "isSelected": false,
      "value": 4,
    },
    Object {
      "appearance": "item",
      "isSelected": false,
      "value": 5,
    },
    Object {
      "appearance": "item",
      "isSelected": false,
      "value": 6,
    },
    Object {
      "appearance": "item",
      "isSelected": false,
      "value": 7,
    },
    Object {
      "appearance": "item",
      "isSelected": false,
      "value": 8,
    },
    Object {
      "appearance": "item",
      "isSelected": false,
      "value": 9,
    },
    Object {
      "appearance": "overflow-end",
      "isSelected": false,
      "value": "...",
    },
    Object {
      "appearance": "item",
      "isSelected": false,
      "value": 23,
    },
    Object {
      "appearance": "item",
      "isSelected": false,
      "value": 24,
    },
    Object {
      "appearance": "item",
      "isSelected": false,
      "value": 25,
    },
  ],
  "setCurrentPage": [Function],
}
`;

exports[`use pagination returns props correctly ({"boundaryPageCount": 5, "currentPage": 6, "nearPageCount": 1, "totalPageCount": 4}) 1`] = `
Object {
  "currentPage": 6,
  "handleFirstClick": [Function],
  "handleLastClick": [Function],
  "handleNextClick": [Function],
  "handlePreviousClick": [Function],
  "isEnd": false,
  "isStart": false,
  "items": Array [
    Object {
      "appearance": "overflow-start",
      "isSelected": false,
      "value": "...",
    },
    Object {
      "appearance": "item",
      "isSelected": false,
      "value": 1,
    },
    Object {
      "appearance": "item",
      "isSelected": false,
      "value": 2,
    },
    Object {
      "appearance": "item",
      "isSelected": false,
      "value": 3,
    },
    Object {
      "appearance": "item",
      "isSelected": false,
      "value": 4,
    },
  ],
  "setCurrentPage": [Function],
}
`;
